src/Form/Field.php

```diff
@@ -255,6 +255,30 @@ public function isRequired(): bool
         return $this->required ?? false;
     }
 
+    protected function isSaveable(): bool
+    {
+        if ($this->isRequired() === true && $this->save() === true && $this->isEmpty() === true) {
+            // check the data of the relevant fields if there is a when option
+            if (empty($this->when) === false && is_array($this->when) === true) {
+                $input = $this->attrs['input'] ?? [];
+
+                if (empty($input) === false) {
+                    foreach ($this->when as $field => $value) {
+                        // if the input data doesn't have data for when fields
+                        // or doesn't match, that means is not required and return true
+                        if (($input[$field] ?? '') !== $value) {
+                            return true;
+                        }
+                    }
+                }
+            }
+
+            return false;
+        }
+
+        return true;
```

Comment:
As the `if` above needs all three conditions to be `true`, this line will make the method return `true` if `$this->save()` is set to `false` – in which case the field is definitely not saveable. But we actually don't want to know if the field can be saved, but if it fulfills the criteria for the `required` option.

What about `needsValue()` as the method name? The return values would of course need to be swapped.

Could you please add a docblock to this method as well? I know the other methods don't have them, but the goal is to add them to all methods over time.

Reply:
`needsValue()` is more accurate name 👍